scripts/common.sh

```diff
@@ -1,49 +1,150 @@
 #!/bin/bash
 
+red="\033[0;31m"
+green="\033[0;32m"
+cyan="\033[0;36m"
+bold="\033[1m"
+nocolor="\033[0m"
+
 scripts_folder=$(dirname "$0")
 
+browser="echo"
+if which firefox > /dev/null 2>&1
+then
+  browser=firefox
+elif which xdg-open > /dev/null 2>&1
+then
+  browser=xdg-open
+elif which open > /dev/null 2>&1
+then
+  browser=open
+fi
+
+EDITOR="${EDITOR-$VISUAL}"
+
+#-------------------------------------------------------------------------------
+function need() {
+  req="$1"
+
+  if [ -z "$(which "$req")" ]; then
```

Comment:
More of an FYI than a `nit`, and feel free to ignore: this can be implemented with bash builtins like `type` or `command` instead of using the `which` binary, which I think makes things a little bit cleaner:

```suggestion
  if ! type -t "$req" &>/dev/null; then
```

The only difference is that (as opposed to `which`) this will also return true for bash functions, aliases, and builtins, which might not be your desired behavior.